scripts/variantstore/wdl/extract/extract_subpop.py

```diff
@@ -1,22 +1,47 @@
 import csv
 import argparse
 
-def extract_subpopulation(input_path, output_path):
-  with open(input_path, newline='') as tsvin, open(output_path, 'w', newline='') as csvout:
-    tsvin = csv.reader(tsvin, delimiter='\t')
-    csvout = csv.writer(csvout, delimiter='\t')
+expected_subpopulations = [
+ "afr",
+ "amr",
+ "eas",
+ "eur",
+ "mid",
+ "oth",
+ "sas"
+]
 
+def extract_subpopulation(input_path):
+  seen_subpopulations = []
+  with open(input_path, newline='') as tsvin:
+    tsvin = csv.reader(tsvin, delimiter='\t')
+    next(tsvin) # skip the header
     for row in tsvin:
-      csvout.writerow([row[0], row[4]])
-
+      row_subpopulation = row[4]
+      if row_subpopulation in expected_subpopulations:
+        if row_subpopulation in seen_subpopulations:
+          # open the correct output file
+          output_file = open(row_subpopulation + "_subpopulation.args", 'a', newline='')
+          csvout = csv.writer(output_file, delimiter='\t')
+          # add sample name to the subpopulation file
+          csvout.writerow([row[0]])
+        else:
+          seen_subpopulations.append(row_subpopulation)
+          # Create a new file for this subpopulation
+          output_file = open(row_subpopulation + "_subpopulation.args", 'w', newline='')
+          csvout = csv.writer(output_file, delimiter='\t')
+          # add sample name to the subpopulation file
+          csvout.writerow([row[0]])
+          print(seen_subpopulations)
+      else:
+        print("WARNING: This list had an unexpected subpopulation", row_subpopulation)
+      ## TODO do I need to close these?
```

Comment:
I think you do need to close them.
If you use a `with` clause then you don't, but I don't think that will work within this looping structure